kythe/go/extractors/config/extractor.go

```diff
@@ -67,25 +68,23 @@ func LocalCopier(repoPath string) func(ctx context.Context, outputDir string) er
 				return nil
 			}
 			if filepath.HasPrefix(path, gitDir) {
-				// Also skip all that .git stuff.
-				return nil
+				return filepath.SkipDir
 			}
 			rel, err := filepath.Rel(repoPath, path)
 			if err != nil {
 				return err
 			}
 			outPath := filepath.Join(outputDir, rel)
 			if info.IsDir() {
```

Comment:
Another approach we could take here is to skip directories entirely, and just use `os.MkdirAll` on any regular file path:

```
if info.Mode().IsRegular() {
  if err := os.MkdirAll(filepath.Dir(outPath), 0755); err != nil {
    return …
  }
  …copy the file…
} else if !info.IsDir() {
  log.Printf("Skipping non-regular file %q", path)
}
return nil
```

or words to that effect?

Reply:
Ah yea good point.  That makes it a bit cleaner.